src/api/controller/api/hiddenResource.js

```diff
@@ -0,0 +1,22 @@
+import Koa from 'koa';
+import route from 'koa-route';
+import fs from 'fs';
+import { unlinkFile } from '../../services/fsHelpers';
+
+const app = new Koa();
+
+const getExportHiddenResources = async ctx => {
+    const hiddenResources = await ctx.hiddenResource.findAll();
+    const date = new Date().getTime();
+    const filePath = `/tmp/hiddenResources${date}.json`;
+    const stream = fs.createWriteStream(filePath);
+    stream.write(JSON.stringify(hiddenResources));
+    stream.end();
+    stream.on('finish', () => unlinkFile(filePath));
+    ctx.body = fs.createReadStream(filePath);
+    ctx.status = 200;
+};
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏We should wait for the stream to finish writing before returning a 200 status code. Currently there is no link between the Promise resolution status of the function and the stream completion.

Looking at existing code, I found a helper function called `streamEnd`, used in `src/api/controller/api/dump.js`, that could probably help here.

EDIT: Actually, looking at this again, I don't even understand why you need to write into a file in the first place. Can't you simply feed `ctx.body` with `JSON.stringify(hiddenResources)`? :thinking: